### PR TITLE
[Pattern Draft] governance-based-project-setup

### PR DESCRIPTION
**Pull request by MaineC**
As discussed in Slack previously this adds a new pattern to link our Governance Levels with information on what pre-requisits need to be fulfilled to reach the respective governance level.

The pattern looks at several factors:
* Knowledge needed by contributors and host team.
* General advise on project structure.
* Maturity levels that need to be reached.
* Relevant patterns.

### CODE REVIEW

**Comment by michael-basil** [comment]
This pattern resonates.   So from an atomic and or reductionist view it makes sense to stand on it's own.  It is useful on it's own and many groups would or could benefit.

I also see stretch opportunities (taking a step back to an integrated or holistic view) with patterns that I/we (from Dojo Center) have proposed and that are in draft.

- #696 Circle Communities
- #695 Agile InnerSource Dojo 

Stretch conversations can look orthogonal so I don't mean to necessarily unpack any intersection opportunities here.

Given a future professional opportunity as an influencer, director or coach I would consider leveraging the patterns above together.   

TL/DR Very supportive and appreciative

**Comment by MaineC** [comment]
I fully agree that we need another pattern with a more holistic view to go along with this one. 

Thanks for your feedback, glad you find it helpful.

File: patterns/1-initial/governance-based-project-setup.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+## Title
+
+Governance Level Guided Project Setup
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md:3]: Found that title hard to understand. Need to think about a suggestion for a better one.

File: patterns/1-initial/governance-based-project-setup.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+## Patlet
+
+A team wants to setup a new InnerSource project or convert an existing project to be InnerSource ready. They already decided on a [Governance Level](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/governance-levels.md) to use. However the team is at a loss at what the chosen Governance Level means in terms of impact on the project.
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md]:
```suggestion
A team wants to a project to be InnerSource ready. They already decided on a [Governance Level](../1-initial/governance-levels.md) to use. However the team is at a loss at what the chosen Governance Level means in terms of impact on the project.
```

I would leave the distinction between new or existing project away, unless this makes a significant difference for this pattern.

Also make the link relative :)

File: patterns/1-initial/governance-based-project-setup.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+## Title
+
+Governance Level Guided Project Setup
+
+## Patlet
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md:5]: We typically sugggest to make the Patlet 2 sentences:
- 1st sentence for the problem
- 2nd setence for the solution

The current Patlet uses a lot more than that. Wondering if we can somehow shorten it?

File: patterns/1-initial/governance-based-project-setup.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+* [Base Documentation](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/base-documentation.md)
+* [Communiation Tooling](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/communication-tooling.md)
+* [Issue Tracker](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/issue-tracker.md)
+* [Praise Participants](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/praise-participants.md)
+* [Standard Release Process](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/release-process.md)
+* [Trusted Committer](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/trusted-committer.md)
+* [Improve Findability](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/improve-findability.md)
+
+
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md:62]:
```suggestion
```

File: patterns/1-initial/governance-based-project-setup.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+For the maturity model, the relevant levels needed are at least:
+
+PP-3, DP-3, DC-3, RS-3, ST-3, CF-3, LS-3, OF-3, CB-3, SP-3, PS-3, RW-3, MP-3, SM-3, CL-3, RO-3
+
+For InnerSource patterns, start by looking at the following ones (in addition to the ones above):
+
+* [Group Support](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/group-support.md)
+* [Explicit Shared Ownership](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/explicit-shared-ownership.md)
+* [Modular Code](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/modular-code.md)
+* [Source Repo different from deployment chain](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/shared-code-repo-different-from-build-repo.md)
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md]:
```suggestion
* [Group Support](../2-structured/group-support.md)
* [Explicit Shared Ownership](../1-initial/explicit-shared-ownership.md)
* [Modular Code](../1-initial/modular-code.md)
* [Source Repo different from deployment chain](../1-initial/shared-code-repo-different-from-build-repo.md)
```

File: patterns/1-initial/governance-based-project-setup.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+* [Review Committee](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/review-committee.md)
+* [Service vs. Library](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/service-vs-library.md)
+* [Transparent cross team decision making](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/transparent-cross-team-decision-making-using-rfcs.md)
+* [Cross team retrospectives](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/cross-team-retrospectives.md)
+* [Incentive Mechanisms](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/incentive-mechanisms-for-voluntary-contribution.md)
+* [Include Product Owners](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/include-product-owners.md)
+* [Reluctance to accept contributions](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/reluctance-to-accept-contributions.md)
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md]:
```suggestion
* [30-day warranty](../2-structured/30-day-warranty.md)
* [Extensions for Sustainable Growth](../2-structured/extensions-for-sustainable-growth.md)
* [Review Committee](../2-structured/review-committee.md)
* [Service vs. Library](../2-structured/service-vs-library.md)
* [Transparent cross team decision making](../2-structured/transparent-cross-team-decision-making-using-rfcs.md)
* [Cross team retrospectives](../1-initial/cross-team-retrospectives.md)
* [Incentive Mechanisms](../1-initial/incentive-mechanisms-for-voluntary-contribution.md)
* [Include Product Owners](../1-initial/include-product-owners.md)
* [Reluctance to accept contributions](../1-initial/reluctance-to-accept-contributions.md)
```

File: patterns/1-initial/governance-based-project-setup.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+For InnerSource patterns, start by looking at the following ones:
+
+* [Base Documentation](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/base-documentation.md)
+* [Communiation Tooling](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/communication-tooling.md)
+* [Issue Tracker](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/issue-tracker.md)
+* [Praise Participants](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/praise-participants.md)
+* [Standard Release Process](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/release-process.md)
+* [Trusted Committer](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/2-structured/trusted-committer.md)
+* [Improve Findability](https://github.com/InnerSourceCommons/InnerSourcePatterns/blob/main/patterns/1-initial/improve-findability.md)
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md]:
```suggestion
* [Base Documentation](../2-structured/base-documentation.md)
* [Communiation Tooling](../2-structured/communication-tooling.md)
* [Issue Tracker](../2-structured/issue-tracker.md)
* [Praise Participants](../2-structured/praise-participants.md)
* [Standard Release Process](../2-structured/release-process.md)
* [Trusted Committer](../2-structured/trusted-committer.md)
* [Improve Findability](../1-initial/improve-findability.md)
```

**Review by spier — COMMENTED** [review]
Thank you @MaineC for pushing your experiences into the open, in this case in the form of patterns!

I am just leaving some initial comments, and will do a more wholistic review at a later point.

Will already commit mini fixes that are purely formatting etc

File: patterns/1-initial/governance-based-project-setup.md (hunk 1 above)
Comment (MaineC) [reply patterns/1-initial/governance-based-project-setup.md:3]: I fully agree - and: naming is hard :)

Hmm. I think I need help brainstorming here.

Something that puts "capabilities" into the title?

What this pattern does is to pull several separate patterns together into a subsystem. I'm sure there is a term for that kind of text. Maybe @kjstol or @NewMexicoKid can provide some information for that?

File: patterns/1-initial/governance-based-project-setup.md (hunk 3 above)
Comment (MaineC) [reply patterns/1-initial/governance-based-project-setup.md:5]: I tried to shorten the patlet further, I don't get it down below three sentences.

**Comment by MaineC** [comment]
Thanks for the initial review @spier - really appreciate the input on this one.

File: patterns/1-initial/governance-based-project-setup.md (hunk 1 above)
Comment (MaineC) [reply patterns/1-initial/governance-based-project-setup.md:3]: Maybe something like "governance level drives maturity needs"?

File: patterns/1-initial/governance-based-project-setup.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+Following our list of best practices, recommended patterns and maturity levels to reach helps address this problem.
+
+## Problem
+
+A team has decided that they want to publish an InnerSource project. They are trying to decide on a governance level. However lacking practical experience they need further guidance on the impact each of the levels will have on their daily doing.
+
+## Context
+
+- The team has expierence working as contributors to other InnerSource projects.
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md]:
```suggestion
- The team has experience working as contributors to other InnerSource projects.
```

Not sure why the spell check GHA did not catch this automatically.

**Comment by MaineC** [comment]
@spier I believe I addressed your comments. Is there anything left to change before this can be merged?

File: patterns/1-initial/governance-based-project-setup.md (hunk 1 above)
Comment (spier) [reply patterns/1-initial/governance-based-project-setup.md:3]: We can improve the title later. Maybe once we find further adopters of the pattern.

File: patterns/1-initial/governance-based-project-setup.md (hunk 3 above)
Comment (spier) [reply patterns/1-initial/governance-based-project-setup.md:5]: I have rewritten the Patlet. What do you think?

File: patterns/1-initial/governance-based-project-setup.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+Definition:
+> People outside the core development team may use the code. They can submit feature requests and bug reports for things they would like to see changed.
+
+To reach this level, a team needs to make their source code readable. They need to give write access to their issue tracker.
+
+Ideally the host team goes through the [InnerSource Learning Path](https://innersourcecommons.org/learn/learning-path/) to get a first understanding of the concepts and roles involved.
+
+The relevant levels of the [maturity model](../2-structured/maturity-model.md) are at least:
+
+PP-1, DP-1, DC-1, RS-1, ST-1, CF-1, LS-1, OF-2, CB-2, SP-2, PA-2, RW-1, MP-1, SM-1, CL-1, RO-1
```
Comment (spier) [inline patterns/1-initial/governance-based-project-setup.md:49]: It feels impractical for the readers to take the short codes from here, and look them up in the maturity model.

Unfortunately I don't have a better idea yet either. Maybe our publishing tooling (gitbook) has a solution for this. We can look at this again once we aim to get this pattern published in our book.

**Review by spier — APPROVED** [review]
@MaineC I have finally had time to fully digest your pattern here.

Only now I have actually understood what the pattern is trying to do, I think :) I have added some explanations at the beginning of the Solution section, describing my understanding if your approach.

I did wonder shortly if the content of this pattern could be merged into the "Governance Levels" pattern itself. Not sure about the pros-cons of this yet.

However as this PR has been sitting idle here for so long already, I suggest to get this merged in our repo, and decide at a later point how we want to proceed with this.

Having this fully integrated into our repo will make it easier to get more eyeballs by other readers on it.

Thank you for documenting your practical InnerSource experiences, and sharing these with us! And sorry that this review took so long

File: patterns/1-initial/governance-based-project-setup.md (hunk 3 above)
Comment (MaineC) [reply patterns/1-initial/governance-based-project-setup.md:5]: The rewrite sounds really good to me.

File: patterns/1-initial/governance-based-project-setup.md (hunk 9 above)
Comment (MaineC) [reply patterns/1-initial/governance-based-project-setup.md:49]: I'm also quite unhappy that we can't sort of link out directly. I included the information here because I believe, it's still valuable to have that link to the maturity model steps.

**Comment by MaineC** [comment]
> @MaineC I have finally had time to fully digest your pattern here.

Thanks so much for taking that time! (And: Happy New Year!)

> Only now I have actually understood what the pattern is trying to do, I think :)
> I have added some explanations at the beginning of the Solution section, 
> describing my understanding if your approach.

What you added sounds correct!


> I did wonder shortly if the content of this pattern could be merged into the
> "Governance Levels" pattern itself. Not sure about the pros-cons of this yet.

The reason I didn't do that is the length of the patterns: I wanted to keep Governance Levels very brief and concise so it's easy to read and understand. Also I wanted to keep focus there on the business decision itself without clouding the discussion with implementation details.

> However as this PR has been sitting idle here for so long already, I suggest 
> to get this merged in our repo, and decide at a later point how we want to proceed with this.
> Having this fully integrated into our repo will make it easier to get more eyeballs by other readers on it.

Sounds reasonable to me. :)


> Thank you for documenting your practical InnerSource experiences, and 
> sharing these with us! And sorry that this review took so long

No worries - glad to see this pattern making progress.

**Review by spier — APPROVED** [review]
I am leaving two comments open, that we may use as input when trying further improve this pattern in the future.
